service/src/main.rs

```diff
@@ -782,3 +789,33 @@ fn bootstrap_funds_from_alice(
 	api.signer = signer_orig;
 	Ok(())
 }
+
+/// In case this is the first validateer on the specified parentchain, we need to trigger
+/// the parentchain block import up until the block where we have registered ourselves.
+fn bootstrap_parentchain_blocks_for_primary_validateer<E: EnclaveBase + TeerexApi + Sidechain>(
+	node_api: &Api<sr25519::Pair, WsRpcClient>,
+	enclave_api: Arc<E>,
+	last_synced_header: &Header,
+	registry_block_hash: Option<Hash>,
+) -> Result<Header, Error> {
+	let registry_header: Header =
+		node_api.get_header(registry_block_hash)?.ok_or(Error::EmptyValue)?;
+	let mut last_synced_header = last_synced_header.clone();
+	let enclave_count_of_previous_block =
+		node_api.enclave_count(Some(*registry_header.parent_hash()))?;
+
+	if enclave_count_of_previous_block == 0 {
+		info!(
+			"We're the first validateer to be registered, syncing parentchain blocks until the one we have registered ourselves on."
+		);
+		let parentchain_block_syncer =
+			ParentchainBlockSyncer::new(node_api.clone(), enclave_api.clone());
```

Comment:
According to feedback above, I would prefer this outside of the function

Reply:
Done. Introduced a new Trait accordingly :)